axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/AbstractFixedAssetLineServiceImpl.java

```diff
@@ -232,9 +278,25 @@ public void filterListByStatus(List<FixedAssetLine> fixedAssetLineList, int stat
       fixedAssetLineList.stream()
           .filter(line -> line.getStatusSelect() == status)
           .forEach(line -> linesToRemove.add(line));
-      fixedAssetLineList.removeIf(line -> line.getStatusSelect() == status);
     }
-    clear(linesToRemove);
+    clear(fixedAssetLineList, linesToRemove);
+  }
+
+  @Override
+  public void filterListByDate(List<FixedAssetLine> fixedAssetLineList, LocalDate date) {
+
+    List<FixedAssetLine> linesToRemove = new ArrayList<>();
+    if (CollectionUtils.isEmpty(fixedAssetLineList) || date == null) {
+      return;
+    }
+    fixedAssetLineList.stream()
+        .filter(
+            line -> line.getDepreciationDate() == null || line.getDepreciationDate().isAfter(date))
+        .forEach(
+            line -> {
+              linesToRemove.add(line);
+            });
```

Comment:
Please directly assign the result of the stream to `linesToRemove` instead of adding lines in a foreach loop